src/handler.py

```diff
@@ -27,26 +27,23 @@
     """
 
 
-async def text_handler(update: Update, context: ContextTypes.DEFAULT_TYPE, con: psycopg2.connect) -> None:
-    _ = context
-    if update.message is None:
-        return
-
-    if update.message.from_user is None:
+async def store_message(message: Message, con: psycopg2.connect) -> None:
```

Comment:
Either it doesn't need to be async, or the db io needs to be moved into a worker thread. E.g., use `asyncio.to_thread()`. A standard way of doing it is to rename this function into `_store_message()`, and then make an async wrapper like:

```python
async def store_message(...) -> str:
    return await asyncio.to_thread(_store_message, ...)
```

This way you have proper async io without blocking the main loop (which is deadly).

Reply:
Or you can use something like aiopg, which is much cleaner